### PR TITLE
remove ES_ENV_SETTINGS from HQ

### PR DESCRIPTION
**Pull request by snopoke**
## Summary
commcare-cloud always sets localsettings.ES_SETTINGS so the values hard coded
here always get overridden.

You can still set env specific settings in the `ES_INDEX_SETTINGS` dict. These will get overridden by values from localsettings if they exist.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/es_utils.py [hunk 1]
```diff
@@ -102,15 +92,9 @@ def meta(self):
             ES_INDEX_SETTINGS.get(settings.SERVER_ENVIRONMENT, {}).get(self.hq_index_name, {})
         )
 
-        overrides = copy(ES_ENV_SETTINGS)
         if settings.ES_SETTINGS is not None:
-            overrides.update({settings.SERVER_ENVIRONMENT: settings.ES_SETTINGS})
-
-        for hq_index_name in ['default', self.hq_index_name]:
-            for key, value in overrides.get(settings.SERVER_ENVIRONMENT, {}).get(hq_index_name, {}).items():
-                if value is REMOVE_SETTING:
```
Comment (dannyroberts) [inline corehq/ex-submodules/pillowtop/es_utils.py:111]: Don't we still want this `REMOVE_SETTING` logic? I think its intent is to let environments remove certain settings from the defaults in `ES_INDEX_SETTINGS`.

Comment (snopoke) [reply corehq/ex-submodules/pillowtop/es_utils.py:111]: I wasn't sure if we wanted to keep this. I think you added it a while ago so if you're happy to remove I'll get rid of it.

Comment (snopoke) [reply corehq/ex-submodules/pillowtop/es_utils.py:111]: sorry, mis-understood you're comment. I'll add this back.

Comment (snopoke) [reply corehq/ex-submodules/pillowtop/es_utils.py:111]: 7b392463dac

**Review by dannyroberts — APPROVED** [review]

File: corehq/ex-submodules/pillowtop/es_utils.py (hunk 1 above)
Comment (dannyroberts) [reply corehq/ex-submodules/pillowtop/es_utils.py:111]: Thanks!

**Pull request merged**